rskj-core/src/main/java/co/rsk/pcc/bto/GetMultisigScriptHash.java

```diff
@@ -0,0 +1,127 @@
+            "getMultisigScriptHash",
+            new String[]{"uint8", "bytes[]"},
+            new String[]{"bytes"}
+    );
+
+    private final int COMPRESSED_PUBLIC_KEY_LENGTH = 33;
+    private final int UNCOMPRESSED_PUBLIC_KEY_LENGTH = 65;
+
+    // Enforced by the 520-byte size limit of the redeem script
+    // (see https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki#520byte_limitation_on_serialized_script_size)
+    private final int MAXIMUM_ALLOWED_SIGNATURES = 15;
+
+    public GetMultisigScriptHash(ExecutionEnvironment executionEnvironment) {
+        super(executionEnvironment);
+    }
+
+    @Override
+    public CallTransaction.Function getFunction() {
+        return function;
+    }
+
+    @Override
+    public Object execute(Object[] arguments) {
+        int minimumSignatures = ((BigInteger) arguments[0]).intValueExact();
+        Object[] publicKeys = (Object[]) arguments[1];
+
+        if (minimumSignatures == 0) {
+            throw new NativeContractIllegalArgumentException("Minimum required signatures must be greater than zero");
+        }
+
+        if (publicKeys.length == 0) {
+            throw new NativeContractIllegalArgumentException("At least one public key is required");
+        }
+
+        if (publicKeys.length < minimumSignatures) {
+            throw new NativeContractIllegalArgumentException(String.format(
+                    "Given public keys (%d) are less than the minimum required signatures (%d)",
+                    publicKeys.length, minimumSignatures
+            ));
+        }
+
+        if (publicKeys.length > MAXIMUM_ALLOWED_SIGNATURES) {
+            throw new NativeContractIllegalArgumentException(String.format(
+                    "Given public keys (%d) are more than the maximum allowed signatures (%d)",
+                    publicKeys.length, MAXIMUM_ALLOWED_SIGNATURES
+            ));
+        }
+
+        Arrays.stream(publicKeys).forEach(o -> {
+            byte[] publicKey = (byte[]) o;
+            if (publicKey.length != COMPRESSED_PUBLIC_KEY_LENGTH && publicKey.length != UNCOMPRESSED_PUBLIC_KEY_LENGTH) {
+                throw new NativeContractIllegalArgumentException(String.format(
+                        "Invalid public key length: %d", publicKey.length
+                ));
+            }
+        });
+
+        List<BtcECKey> btcPublicKeys = Arrays.stream(publicKeys)
+                .map(o -> {
+                    try {
+                        return BtcECKey.fromPublicOnly((byte[]) o);
+                    } catch (IllegalArgumentException e) {
+                        throw new NativeContractIllegalArgumentException(String.format(
+                                "Invalid public key format: %s", Hex.toHexString((byte[]) o)
+                        ), e);
+                    }
+                })
+                .map(k -> BtcECKey.fromPublicOnly(k.getPubKeyPoint().getEncoded(true)))
```

Comment:
If you want to convert the uncompresed to compressed i thinks its better doing it on the previous step while we validate the size of the public key, then we could remove the `map(k -> BtcECKey.fromPublicOnly(k.getPubKeyPoint().getEncoded(true)))`  as you already have the BtcECKey from the previous map.

Reply:
What exactly do you mean changing?